lib/block_ccm.mli

```diff
@@ -1,9 +1,12 @@
-module Make (B: Mirage_types.BLOCK
-             with type 'a io = 'a Lwt.t
-              and type page_aligned_buffer = Cstruct.t) : sig
-  include Mirage_types.BLOCK
-    with type 'a io = 'a Lwt.t
-     and type page_aligned_buffer = Cstruct.t
-
-  val connect : ?maclen:int -> ?nonce_len:int -> key:Cstruct.t -> B.t -> t io
+module Make (B: Mirage_block_lwt.S) : sig
+
+  type error = private [> Mirage_device.error | `DecryptError ]
```

Comment:
(but I'm still not sure whether this is correct... it get's (as argument) a `B`, and shouldn't the error contain `B.error | [whatever]`?  (but then, I get that error that B.error is not a variant type)...